graphql_compiler/schema_transformation/split_query.py

```diff
@@ -491,23 +637,32 @@ def _split_vertex_fields_intra_and_cross_schema(
     return intra_schema_fields, cross_schema_fields
 
 
-def _get_selections_from_property_and_vertex_fields(property_fields_map, vertex_fields):
+def _get_selections_from_property_and_vertex_fields(
+    # OrderedDict is unsubscriptable (pylint E1136)
+    property_fields_map: "OrderedDict[str, FieldNode]",
+    vertex_fields: List[FieldNode],
+) -> List[FieldNode]:
```

Comment:
Could this return `List[SelectionNode]` and maybe take `OrderedDict[str, SelectionNode]` and `Sequence[SelectionNode]`?

If we can't change the input types easily, could we add a runtime `isinstance` check on all the things we're adding to the `selections` list to make sure they are all `SelectionNode`? That way the input types can stay the same, the return type can become what we need it to be downstream, and we only pay a small runtime cost that doesn't increase memory pressure.